metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Taxonomy.java

```diff
@@ -76,16 +105,86 @@ public long getTaxonomyId() {
         return taxonomyId;
     }
 
+    public void setTaxonomyId(long taxonomyId) {
+        this.taxonomyId = taxonomyId;
+    }
+
     public String getName() {
         return name;
     }
 
+    public void setName(String name) {
+        this.name = name;
+    }
+
     /**
      * Release date control: taxonomies are always public.
      */
     @Override
+    @JsonIgnore
     public LocalDate getReleaseDate() {
         return null;
     }
 
+    public Rank getRank() {
+        return rank;
+    }
+
+    public void setRank(Rank rank) {
+        this.rank = rank;
+    }
+
+    @ApiModelProperty(position = 5, readOnly = true)
+    public Long getTaxonomySpeciesId() {
+        return (taxonomySpecies == null) ? -1 : taxonomySpecies.getTaxonomyId();
+    }
+
+    @ApiModelProperty(position = 7, readOnly = true)
+    public Long getTaxonomyGenusId() {
+        return (taxonomyGenus == null) ? -1 : taxonomyGenus.getTaxonomyId();
+    }
+
+    @ApiModelProperty(position = 9, readOnly = true)
+    public Long getTaxOrderId() {
+        return (taxonomyOrder == null) ? -1 : taxonomyOrder.getTaxonomyId();
+    }
+
+    @ApiModelProperty(position = 11, readOnly = true)
+    public Long getTaxonomyClassId() {
+        return (taxonomyClass == null) ? -1 : taxonomyClass.getTaxonomyId();
+    }
```

Comment:
Are these properties required? They are not intended to be used directly anyway, but are just for searching purposes instead. I. e. there is no scenario where someone would want to retrieve class ID **directly** from the Taxonomy entity. If I understand correctly, search by class/order/etc. is implemented by a column join, so it should still work without those entities, right?

Reply:
yes it does wordk , i have just put to show the user the taxonomyId of the parents directly instead of traversing through the url , this is not in our usecase but thought it would be handy for viewing

Reply:
But then there's additional information to be represented for easier viewing, for example names of genus/order/class etc. would also be convenient. The problem is that it creates lots of code we have to maintain; e. g. if in the future we decide to add an additional level for fixed-rank taxonomies, we will have to update this part as well. Since the API is intended for programmatic access, the ease of viewing isn't the priority

Reply:
ya, I understand, then it would be easier for me to delete this extra code then :)